### PR TITLE
UserPageBuster improvements

### PR DESCRIPTION
**Pull request by aleclearmind**
* Move userpagebuster-related stuff in its own directory
* Simplify build (using `__attribute__((constructor))` instead of invoking `ld` directly)
* Introduce a simple working example and demo instructions to assess success

### CODE REVIEW

File: userpagebuster/Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+example: example.c
+	@ ${CC} $< -g -o $@
+
+userpagebuster.so: userpagebuster.c
+	@ ${CC} -shared -fPIC ${GLIB} $< -ldl -o $@
+
+clean:
+	@ echo "Cleaning up ..."
+	@ rm *.so *.o
```
[the remark this answers is not in the document]
Comment (aleclearmind) [reply userpagebuster/Makefile]: No. Dropped.

File: userpagebuster/Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# make		# build library
+# make clean	# remove ALL binaries and objects
+
+.PHONY: all clean
+
+CC := gcc
+
+GLIB = $(shell pkg-config --cflags --libs glib-2.0)
+CFLAGS = -fPIC
+LDFLAGS = -shared -ldl
```
Comment (zTehRyaN) [inline userpagebuster/Makefile:11]: I think this no more needed

File: userpagebuster/Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+all: userpagebuster.so example
+
+example: example.c
+	@ ${CC} $< -g -o $@
+
+userpagebuster.so: userpagebuster.c
+	@ ${CC} -shared -fPIC ${GLIB} $< -ldl -o $@
```
Comment (zTehRyaN) [inline userpagebuster/Makefile:19]: Why don't use ${CFLAGS}?

File: userpagebuster/Makefile (hunk 2 above)
Comment (aleclearmind) [reply userpagebuster/Makefile]: Gone.

File: userpagebuster/Makefile (hunk 3 above)
Comment (aleclearmind) [reply userpagebuster/Makefile:17]: No big difference, I like to use variables if they are used in multiple place or if they have an important semantic.
Here, I just inlined it. Let me know if you want to revert the change.

Comment (zTehRyaN) [reply userpagebuster/Makefile:17]: Ok good!

**Pull request merged**